egs/swbd/s5c/local/nnet3/run_tdnn_3dcnn.sh

```diff
@@ -0,0 +1,149 @@
+# --num-threads 16 and --minibatch-size 128.
+
+stage=9
+affix=
+train_stage=-10
+has_fisher=true
+speed_perturb=true
+use_cnn=true
+cmvn_opts="--norm-means=false --norm-vars=false"
+common_egs_dir=
+reporting_email=
+remove_egs=true
+
+. cmd.sh
+. ./path.sh
+. ./utils/parse_options.sh
+
+
+if ! cuda-compiled; then
+  cat <<EOF && exit 1
+This script is intended to be used with GPUs but you have not compiled Kaldi with CUDA
+If you want to use GPUs (and have them), go to src/, and configure and make on a machine
+where "nvcc" is installed.
+EOF
+fi
+
+suffix=
+if [ "$speed_perturb" == "true" ]; then
+  suffix=_sp
+fi
+
+cnn_opts=()
+if [ "$use_cnn" == "true" ]; then
+# When the cnn option is on, the LDA is replaced by
+# an CNN layer at the front of the network
+# and the recipe uses fbank features as the CNN input
+# As the dimension of the CNN output is usually large, we place
+# a linear layer at the output of CNN for dimension reduction
+# The ivectors are processed through a fully connected affine layer,
+# then concatenated with the CNN bottleneck output and
+# passed to the deeper part of the network.
+# Due to the data compression issue, it is better to convert MFCC
+# to FBANK in the network instead of directly using FBANK features
+# from the storage. This script uses MFCC features as its input
+# and were converted to FBANK features with an inverse of DCT matrix
+# at the first layer of the network.
+
+  cnn_bottleneck_dim=  # remove this option if you don't want to add the bottleneck affine
+  cepstral_lifter=  # have to fill this in if you are not using the default lifter value in the production of MFCC
+                    # Here we assume your are using the default lifter value (22.0)
+  cnn_layer="--filt-x-dim=4 --filt-y-dim=4 --num-filters=1024"
+  cnn_opts+=(--cnn.layer "$cnn_layer")
+  cnn_layer="--filt-x-dim=2 --filt-y-dim=4 --filt-z-dim=16 --num-filters=1"
+  cnn_opts+=(--cnn.layer "$cnn_layer")
+
+  cnn_layer="--filt-x-dim=1 --filt-y-dim=4 --num-filters=1024"
+  cnn_opts+=(--cnn.layer "$cnn_layer")
+  cnn_layer="--filt-x-dim=1 --filt-y-dim=4 --filt-z-dim=32 --num-filters=1"
+  cnn_opts+=(--cnn.layer "$cnn_layer")
+
+  [ ! -z "$cnn_bottleneck_dim" ] && cnn_opts+=(--cnn.bottleneck-dim $cnn_bottleneck_dim)
+  [ ! -z "$cepstral_lifter" ] && cnn_opts+=(--cnn.cepstral-lifter $cepstral_lifter)
+  affix=3dcnn
+  # We found that using CNN with mean-normalized features obtains better results
+  cmvn_opts="--norm-means=true --norm-vars=false"
+fi
+
+dir=exp/nnet3/tdnn_testminibatch
+dir=$dir${affix:+_$affix}
+dir=${dir}$suffix
+train_set=train_nodup$suffix
+ali_dir=exp/tri4_ali_nodup$suffix
+
+local/nnet3/run_ivector_common.sh --stage $stage \
+	--speed-perturb $speed_perturb || exit 1;
+
+if [ $stage -le 9 ]; then
+  echo "$0: creating neural net configs";
+
+  # create the config files for nnet initialization
+  python steps/nnet3/tdnn/make_configs.py  \
```

Comment:
To simplify the code add two calls to make_configs.py one for normal TDNN and one for CNN. In this case you can directly provide the --cnn-layer options to the make_configs.py script, rather than storing them in another variable cnn_opts.